pkg/xds/generator/zoneproxy/destinations.go

```diff
@@ -63,12 +70,41 @@ func buildMeshServiceDestinations(
 				Mesh:            ms.GetMeta().GetMesh(),
 				DestinationName: ms.DestinationName(port.Port),
 				SNI:             sni,
+				Resource: model.ResolveBackendRef(
+					ms.Meta,
+					model.ResourceToBackendRef(ms, meshservice_api.MeshServiceType, port.Port),
+				),
```

Comment:
If there is already a resource `ms`, why do we need to create BackendRef and then resolve it back to the resource? Can we just do:
```suggestion
				Resource: core_model.ResolvedBackendRef{
					Resource: pointer.To(core_model.NewTypedResourceIdentifier(ms, core_model.WithSectionName(port.GetName()))),
				},
```